doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+| 6     | Event Infra to read Event table parameters (size and # of days) from a config file. |                     |
+| 7     | NBI interface (gNMI and REST) and CLI                                           |                     |
+| 7.1   | Events                                                        	              |                     |
+| 7.1.1 | Openconfig interface to pull event information.                                 |                     |
+| 7.1.2 | Openconfig interface to pull event summary information.                         |                     |
+|       | Event summary information to contain cumulative counters for:                   |                     |
+|       | - Raised-count (events)                                                         |                     |
+| 7.1.3 | Openconfig interface to pull events using following filters                     |                     |
+|       | - ALL ( pull all events)                                                        |                     |
+|       | - Severity.                                                                     |                     |
+|       | - Recent records (eg., last 5 minutes, one hour, one day).                      |                     |
+|       | - Records between two timestamps,  one timestamp and end, and beginning and a timestamp. |                    |
+|       | - All records between two Sequence Numbers (incl begin and end)                 |                     |
+| 7.2   | Alarms                                                        	              |                     |
+| 7.2.1 | Openconfig interface to pull alarm information.                                 |                     |
+| 7.2.2 | Openconfig interface to pull alarm summary information.                         |                     |
+|       | Counters for Total, Critical, Major, Minor, Warning, Acknowledged               |                     |
+| 7.2.3 | Openconfig interface to pull alarms using following filters                     |                     |
+|       | - All (pull all events)                                                         |                     |
+|       | - Severity.                                                                     |                     |
+|       | - Recent alarms (eg., last 5 minutes, one hour, one day).                       |                     |
+|       | - Records between two timestamps, one timestamp and end, and beginning and a timestamp. |                     |
+|       | - All records between two Sequence Numbers (incl end and begin)                 |                     |
+| 7.2.4 | Openconfig interface  to acknowledge an alarm.                                  |                     |
+| 8     | CLI commands                                                                    |                     |
```

Comment:
This has been emphasized in the very first section: 

Note: Wherever CLI is specified, it is the CLISH cli that is referred - SONiC native (CLICK) CLI is not updated for this feature.

Reply:
Unless SONiC moves to work only on top of CLISH I don't understand how this path can be taken.
@lguohan I believe we need a clear definition to avoid such comments for new HLDs to come.

As for this HLD I would recommend to have at least the show commands. The configuration if needed can be done via the config db if needed but the feature is kind of limited if the user need to read the redis db to understand the system status. And it can be very simple with the autogenerated CLI executed manually on top of the yang model.